pkg/storage/interface.go

```diff
@@ -138,7 +138,7 @@ func CopyInstantProfileTree(pt InstantProfileTree) *ProfileTree {
 type InstantProfile interface {
 	ProfileTree() InstantProfileTree
 	ProfileMeta() InstantProfileMeta
-	Samples() map[string]*Sample
+	Samples() map[[16]byte]*Sample
```

Comment:
I prefer this to be a string. A copy happens anyways, so `[16]byte` is no better in my opinion and locks the interface into having to be 16 bytes.

Reply:
We now could also have this be a `Samples() map[uuid.UUID]*Sample` too, what about that one? 
Happy to use the string nonetheless. I guess in the case of a string we don't want to use `uuid.String()` still but cast like before?

Reply:
Yeah, I think using the opaque `[]byte` (and consequently `string` as map keys, this is a bit of a hack but allows for various proven optimization techniques) would be best as we're not locking ourselves into a specific mechanism or length.